### ISSUE TITLE
[Enhancement] Add "Add(IEnumerable<T> collection)" to ObservableRangeCollection<T>

### ISSUE THREAD

**Issue by maxkoshevoi** [xamarin/XamarinCommunityToolkit#756]
## Summary

`ObservableRangeCollection<T>` is a great class it allows it's instances to be readonly and updated though `..Range` methods (without the need to backing field and `SetProperty` in setter).

But there is one minor problem if collection is readonly it cannot be easily initialized in object initializer. 

## API Changes

```cs
/// <summary>
/// To be used in collection initializer
/// </summary>
[EditorBrowsable(EditorBrowsableState.Never)]
public void Add(IEnumerable<T> collection) => AddRange(collection);
```

## Intended Use Case

```cs
EventTypes = new ListViewModel<EventType>()
{
    ItemsSource = { timetableInfo.EventTypes(lessonInfo.Lesson.ID).OrderBy(et => et.Entity.ShortName) }
};
```

## Who Will Do The Work?

- [x] I am willing to take this on myself
- [ ] Just putting this out there for someone to pick up

**Comment by pictos** [comment]
@maxkoshevoi thanks for you suggestion, I know that will make devs life easier, but TBH I'm not sold to add that just to allow ctor initialization since the user can do it by himself with minimum effort.

```csharp
// some method
void Init()
{
			var list = new List<string>
			{
				"a",
				"a",
				"a",
			};

			var x = new ObservableRangeCollection<string>
			{
				list,
				list
			};
}

// some where in your code
	static class ObservableExtension
	{
                // you can use T instead of string
		public static void Add(this ObservableRangeCollection<string> list, IEnumerable<string> collection)
 => list.AddRange(collection);
	}
```

For more info see [chapter 10.2.2 of this book](https://livebook.manning.com/book/c-sharp-in-depth-fourth-edition/chapter-10/89)

**Comment by maxkoshevoi** [comment]
@pictos That's exactly what I've done in my project, but the point of this NuGet is to minimize amount of code that user needs to write when working with Xamarin.Forms app. And duplication this extension method across projects doesn't seems as a good solution. This method wouldn't we visible in IntelliSense.

**Comment by maxkoshevoi** [comment]
@pictos Just to be clear, are you rejecting this proposal and issue should be closed, is another opinion needed on this, or can I create a PR?

**Comment by pictos** [comment]
@maxkoshevoi you can open a PR for that, but instead of adding the add code into the ObservableRangeCollection class as you propose, create a class in the same namespace and create an extension method for that. I can remember that works for me.

**Comment by AndreiMisiukevich** [comment]
Fixed in #891
